flink-clients/src/main/java/org/apache/flink/client/program/MiniClusterClient.java

```diff
@@ -120,7 +135,25 @@ public String stopWithSavepoint(JobID jobId, boolean advanceToEndOfEventTime, @N
 
 	@Override
 	public String getWebInterfaceURL() {
-		return miniCluster.getRestAddress().toString();
+		try {
+			return miniCluster.getRestAddress().get().toString();
+		} catch (InterruptedException | ExecutionException e) {
+			ExceptionUtils.checkInterrupted(e);
+
+			LOG.warn("Could not retrieve the web interface URL for the cluster.", e);
+			return "Unknown address.";
+		}
+	}
+
+	@Override
+	public void shutDownCluster() {
```

Comment:
This seems correct but it is a change of behaviour as previously we were simply throwing an `UnsupportedOperationException`. Given this, it deserves a separate JIRA issue and separate commit.

Reply:
OK. I will add a default "implementation" throw an `UnsupportedOperationException` in this pass.